models/balancer/ethereum/balancer_v2_ethereum_pools_fees.sql

```diff
@@ -0,0 +1,42 @@
+{{
+    config(
+        schema = 'balancer_v2_ethereum',
+        alias='pools_fees',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['block_number', 'tx_hash', 'index'],
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "balancer_v2",
+                                    \'["masquot", "jacektrocinski"]\') }}'
+    ) 
+}}
+
+{% set event_signature = "0xa9ba3ffe0b6c366b81232caab38605a0699ad5398d6cce76f91ee809e322dafc" %}
+
+WITH registered_pools AS (
+    SELECT DISTINCT
+        `poolAddress` AS pool_address
+    FROM
+        {{ source ('balancer_v2_ethereum', 'Vault_evt_PoolRegistered') }}
+)
+SELECT
+    /*+ BROADCASTJOIN (registered_pools) */
+    logs.contract_address,
+    logs.tx_hash,
+    logs.tx_index,
+    logs.index,
+    logs.block_time,
+    logs.block_number,
+    bytea2numeric_v2 (SUBSTRING(logs.data FROM 32 FOR 64)) * 1 AS swap_fee_percentage
```

Comment:
have you noticed any improvement in query speed if we select fewer columns? I guess you mentioned something like this on Discord.

Reply:
Yes, there is significant performance improvement if we return less columns.

If we `SELECT` one or two columns then the query runs in ~1-2 minutes instead of ~6 minutes.

This means, that even with less columns selected, we likely still need to use the incremental loading strategy.

Unless of course ~1-2 minutes is Ok.

For the incremental loading strategy to work we need, at minimum, the columns that are apart of the unique key.

Reply:
I believe `tx_index` and `index` have the same values so you can delete one of them.